### PR TITLE
Add Kerberos-aware Kafka to testing env

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR adds new Docker images and test cases to ensure Kerberos authentication for Kafka is working.

The following elements are added to the test environment:
- `kerberos_kdc`: Kerberos key distribution center
- `kafka_kerberos`: Kerberos-aware Kafka instance

When creating the environment I used [ist-dsi/docker-kerberos](https://github.com/ist-dsi/docker-kerberos) environment as a foundation for our own. I have added it's license to the folder `kerberos_kdc`. I haven't added is to our NOTICE.txt, as it is just a testing "dependency".

## Why is it important?

This way I am confident to remove the Beta tag from Kerberos authentication.

The Kerberos KDC is going to be useful when testing the future Kerberos authentication with Elasticsearch.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works

## Related issues

Parent-issue:  https://github.com/elastic/beats/issues/16565

### CODE REVIEW

File: testing/environments/docker/kafka_kerberos/run.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+wait_for_port 2181
+
+
+echo "Starting Kafka broker"
+mkdir -p ${KAFKA_LOGS_DIR}
+_JAVA_OPTIONS="${_JAVA_OPTIONS} -Djava.security.auth.login.config=/etc/kafka/server_jaas.conf" \
+${KAFKA_HOME}/bin/kafka-server-start.sh ${KAFKA_HOME}/config/server.properties \
+    --override delete.topic.enable=true --override advertised.host.name=${KAFKA_ADVERTISED_HOST} \
+    --override listeners=PLAINTEXT://kafka_kerberos:9092,SASL_SSL://kafka_kerberos:9093 \
+    --override advertised.listeners=PLAINTEXT://kafka_kerberos:9092,SASL_SSL://kafka_kerberos:9093 \
```
Comment (jsoriano) [inline testing/environments/docker/kafka_kerberos/run.sh]: We are setting an `ADVERTISED_HOST` environment variable, wdyt about using it here?
```suggestion
    --override advertised.listeners=PLAINTEXT://${ADVERTISED_HOST}:9092,SASL_SSL://${ADVERTISED_HOST}:9093 \
```

File: libbeat/outputs/kafka/kafka_integration_test.go [hunk 2]
```diff
@@ -47,6 +47,21 @@ const (
...
 )
 
+var (
+	kafkaHost = fmt.Sprintf("%v:%v",
+		getenv("KAFKA_HOST", kafkaDefaultHost),
+		getenv("KAFKA_PORT", kafkaDefaultPort),
+	)
+	kafkaKerberosHost = fmt.Sprintf("%v:%v",
+		getenv("KAFKA_KERBEROS_HOST", kafkaDefaultHost),
+		getenv("KAFKA_PORT", kafkaDefaultPort),
```
Comment (jsoriano) [inline libbeat/outputs/kafka/kafka_integration_test.go]: Kafka secure listener is on port `9093`, how does it work? :thinking:

File: filebeat/input/kafka/config.go [hunk 3]
```diff
@@ -181,7 +180,6 @@ func newSaramaConfig(config kafkaInputConfig) (*sarama.Config, error) {
 	}
 
 	if config.Kerberos != nil {
-		cfgwarn.Beta("Kerberos authentication for Kafka is beta.")
```
[the remark this answers is not in the document]
Comment (kvch) [reply filebeat/input/kafka/config.go]: The changelog entry does not mention that it was in beta. So I don't think we need to add anything.

File: libbeat/docker-compose.yml [hunk 4]
```diff
@@ -130,3 +133,24 @@ services:
...
+
+  kafka_kerberos:
+    build: ${ES_BEATS}/testing/environments/docker/kafka_kerberos
+    expose:
+      - 9092
+      - 2181
+    environment:
+      - ADVERTISED_HOST=kafka_kerberos
+    healthcheck:
+      test: ["CMD", "klist", "-V"]
```
[the remark this answers is not in the document]
Comment (kvch) [reply libbeat/docker-compose.yml]: It's an oversight from me. Thanks for catching it.

File: libbeat/outputs/kafka/kafka_integration_test.go (hunk 2 above)
Comment (kvch) [reply libbeat/outputs/kafka/kafka_integration_test.go]: I configured `9092` in an override when running Kafka in the `run.sh`. Or I am not sure I understand your question.

File: filebeat/input/kafka/config.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsoriano) [reply filebeat/input/kafka/config.go]: Oh, it hasn't been released yet, ok then :+1:

File: libbeat/outputs/kafka/kafka_integration_test.go (hunk 2 above)
Comment (jsoriano) [reply libbeat/outputs/kafka/kafka_integration_test.go]: I refer to this, is Kerberos authentication required in both plaintext and SASL listeners?
```
    --override listeners=PLAINTEXT://kafka_kerberos:9092,SASL_SSL://kafka_kerberos:9093 \
    --override advertised.listeners=PLAINTEXT://${ADVERTISED_HOST}:9092,SASL_SSL://${ADVERTISED_HOST}:9093 \
```

**Pull request closed without merging**